codecov.yml

```diff
@@ -0,0 +1,45 @@
+codecov:
+  branch: main
+  require_ci_to_pass: no
+
+comment:
+  layout: "reach,diff,flags,tree"
+  behavior: default
+  require_changes: no
+  require_base: no
+  require_head: no
+  after_n_builds: 1
+  show_carryforward_flags: true
+
+github_checks:
+  annotations: true
+
+coverage:
+  status:
+    patch:
+      default:
+        target: auto
+        threshold: 5%
+        base: auto
+        only_pulls: true
+        flags:
+          - unit-tests
+    project:
+      default:
+        target: auto
+      theia-unit-tests:
+        target: auto
+        flags:
+          - unit-tests
+
+flag_management:
+  default_rules:
+    carryforward: true
+ignore:
+  - "**/testing/*.go"
+  - "**/mock_*.go"
+  - "**/*generate*.go"
```

Comment:
Don’t need to check for pkg/api as well, no tests are written for the folder yet.